tests_integration/__tests__/line-suffix-boundary.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+/** @type {import('prettier')} */
+const prettier = require("prettier-local");
+
+const {
+  group,
+  indent,
+  line,
+  lineSuffix,
+  lineSuffixBoundary,
+  softline,
+} = prettier.doc.builders;
+
+const printDoc = require("../printDoc");
```

Comment:
We can test this with `prettier.__debug.printDocToString`

Reply:
It's not the same. E.g., `propagateBreaks` isn't called in `__debug.printDocToString`. Even though `propagateBreaks` isn't strictly needed for this specific test, in general it's safer to test docs by printing them the exact way it really happens in production. I hope we'll add more tests for doc printing.